domains/certificates/Query.API/API.UnitTests/DataSyncSyncer/Persistence/SyncStateTest.cs

```diff
@@ -1,48 +1,145 @@
 using System;
-using System.Collections.Generic;
-using API.DataSyncSyncer.Client.Dto;
-using IntegrationEvents;
+using System.Threading;
+using System.Threading.Tasks;
+using API.DataSyncSyncer.Persistence;
+using API.MasterDataService;
+using API.Query.API.Projections;
+using CertificateEvents.Primitives;
+using FluentAssertions;
+using Marten;
+using Microsoft.Extensions.Logging;
+using Moq;
 using Xunit;
 
-namespace API.DataSyncSyncer.Persistence;
+namespace API.UnitTests.DataSyncSyncer.Persistence;
 
 public class SyncStateTest
 {
-    private readonly SyncState syncState;
+    private readonly MasterData masterData = new(
+        GSRN: "gsrn",
+        GridArea: "gridArea",
+        Type: MeteringPointType.Production,
+        Technology: new Technology(FuelCode: "F00000000", TechCode: "T010000"),
+        MeteringPointOwner: "meteringPointOwner",
+        MeteringPointOnboardedStartDate: DateTimeOffset.Now.AddDays(-1));
 
-    public SyncStateTest()
+
+    [Fact]
+    public async Task GetPeriodStartTime_NoDataInStore_ReturnsMeteringPointOnboardedStartDate()
     {
-        syncState = new SyncState();
+        var storeMock = CreateStoreMock(data: null);
+
+        var syncState = new SyncState(storeMock.Object, Mock.Of<ILogger<SyncState>>());
+
+        var actualPeriodStartTime = await syncState.GetPeriodStartTime(masterData);
+
+        actualPeriodStartTime.Should().Be(masterData.MeteringPointOnboardedStartDate.ToUnixTimeSeconds());
     }
 
     [Fact]
-    public void GetPeriodStartTime_GetTime()
+    public async Task GetPeriodStartTime_OneCertificateInStore_ReturnsMeteringPointOnboardedStartDate()
     {
-        var now = DateTimeOffset.Now;
-        var currentStartTime = syncState.GetPeriodStartTime("gsrn", now);
-        Assert.Equal(now.ToUnixTimeSeconds(), currentStartTime);
+        var view = new CertificatesByOwnerView { Owner = masterData.MeteringPointOwner };
+        var start = masterData.MeteringPointOnboardedStartDate;
```

Comment:
Could be nice with some new lines for readability like in `GetPeriodStartTime_TwoCertificatesInStore_ReturnsDataMaxDate()`

Reply:
fixed in 81f2db9fe13461baa97b115a32b88811f11d1a83